### PR TITLE
MF-1190 - Add pkg for library packages

### LINKED ISSUE #1190: Add pkg for library packages
**FEATURE REQUEST**

_2. Describe the feature you are requesting, as well as the possible use case(s) for it._

With the introduction of Transformers instead of Normalizer service, and moving Messaging to a separate package, we started using corresponding packages as libraries. This means that those packages are specifying interfaces and, in some cases, an implementation that will be imported and used as a library in Mainflux and other, external, projects. According to [this](https://github.com/golang-standards/project-layout/tree/master/pkg) project layout proposal, we should move the library code to the `pkg` package. Regardless of this (or any other) layout template, this would bring some benefits such as:
- removing some packages from the project root, making monorepo look cleaner
- better logical organization of the code, separating library code from services

The list of packages that can be moved to the umbrella `pkg` package:
- errors
- messaging
- sdk
- transformers

_3. Indicate the importance of this feature to you (must-have, should-have, nice-to-have)._
This is a should-have feature.

### PR DESCRIPTION
**Pull request by dborovcanin**
### What does this do?
This pull request moves library code to the `pkg` package.

### Which issue(s) does this PR fix/relate to?
This pull request resolves #1190.

### List any changes that modify/break current functionality
Import paths are changed.

### Have you included tests for your changes?
No, this is only moving lib code to the umbrella `pkg` package.

### Did you document any new/modified functionality?
No, because there are no code changes.

### CODE REVIEW

**Review by nmarcetic — DISMISSED** [review]

**Review by nmarcetic — COMMENTED** [review]
CI is failing

**Review by drasko — APPROVED** [review]
LGTM

File: cmd/cli/main.go [hunk 1]
```diff
@@ -7,7 +7,7 @@ import (
 	"log"
 
 	"github.com/mainflux/mainflux/cli"
-	"github.com/mainflux/mainflux/sdk/go"
+	sdk "github.com/mainflux/mainflux/pkg/sdk/go"
```
[the remark this answers is not in the document]
Comment (dborovcanin) [reply cmd/cli/main.go:10]: Let's not address it in this PR. I already updated dependencies, which is also out of scope for this PR.

**Review by manuio — APPROVED** [review]
LGTM

**Pull request merged**